src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,131 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double DEFAULT_MULTIPLICATION_NUMBER = 1.5;
+    private static final int START_ARRAY_SIZE = 0;
+    private T[] elementsData;
+    private int capacity;
+    private int size;
+
+    public ArrayList() {
+        this.elementsData = (T[]) new Object[DEFAULT_CAPACITY];
+        this.capacity = DEFAULT_CAPACITY;
+        this.size = START_ARRAY_SIZE;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size() == capacity) {
+            resizeArray();
+        }
+        elementsData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size() == capacity) {
+            resizeArray();
+        }
+        if (index >= 0 && index <= size) {
+            T[] newElementsData = (T[]) new Object[capacity];
+            for (int i = 0; i < index; i++) {
+                newElementsData[i] = elementsData[i];
+            }
+            newElementsData[index] = value;
+            for (int i = index; i < size(); i++) {
+                newElementsData[i + 1] = elementsData[i];
+            }
+            size++;
+            elementsData = newElementsData;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Index doesn't found");
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index < size && index >= 0) {
+            return elementsData[index];
+        }
+        throw new ArrayListIndexOutOfBoundsException("Index out of bounds for index: " + index);
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index < size && index >= 0) {
+            elementsData[index] = value;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Index doesn't found");
```

Comment:
Use informative exception messages. The message 'Index doesn't found' is not informative enough. It would be better to include the invalid index value and the valid range within the message, for example: 'Index: ' + index + ', Size: ' + size